sources/engine/Stride.Graphics/Direct3D/GraphicsResourceBase.Direct3D.cs

```diff
@@ -108,6 +108,21 @@ internal static SharpDX.Direct3D11.CpuAccessFlags GetCpuAccessFlagsFromUsage(Gra
                 comObject = null;
             }
         }
+
+        /// <summary> Returns true if the object was previously allocated and has been released </summary>
+        internal static bool TryReleaseComObject<T>(ref T comObject) where T : CppObject, IUnknown
+        {
+            if(comObject != null && comObject.NativePointer != IntPtr.Zero)
+            {
+                var refCountResult = comObject.Release();
+                Debug.Assert(refCountResult >= 0);
+                comObject = null;
+                return true;
+            }
+            
+            comObject = null;
+            return false;
+        }
```

Comment:
Added this new signature instead of changing the one above, the comment on that one is cryptic, I don't have the time or patience to see if I break anything by removing it so I'll leave it in.